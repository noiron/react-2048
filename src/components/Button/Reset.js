import React from 'react';
import PropTypes from 'prop-types';
import ResetSvg from '../../assets/svg/reset.svg';
import Button from '../Button';

const Reset = props => {
    return (
        <Button img={ResetSvg} desc="reset" {...props} />
    )
}

Reset.propTypes = {
    handleClick: PropTypes.func.isRequired
}

export default Reset;
