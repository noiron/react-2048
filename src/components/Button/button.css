.button {
    width: .4rem;
    height: .4rem;
    margin-right: .1rem;
    display: inline-block;
    cursor: pointer;
    text-align: center;
    box-shadow: 1px 1px 5px #888888;
    position: relative;
}

.button img {
    position: absolute;
    top: 20%;
    right: 15%;
}