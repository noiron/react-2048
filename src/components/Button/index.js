import React from 'react';
import PropTypes from 'prop-types';
import styles from './button.css';

const Button = (props) => {
    return (
        <div onClick={props.handleClick} className={styles.button}>
            <img src={props.img} alt={props.desc} />
        </div>
    )
}

Button.propTypes = {
    handleClick: PropTypes.func.isRequired,
    img: PropTypes.string.isRequired,
    desc: PropTypes.string.isRequired,
}

export default Button;
