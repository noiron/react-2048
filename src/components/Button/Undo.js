import React from 'react';
import PropTypes from 'prop-types';
import UndoSvg from '../../assets/svg/undo.svg';
import Button from '../Button';

const Undo = props => {
    return (
        <Button img={UndoSvg} desc="undo" {...props} />
    )
}

Undo.propTypes = {
    handleClick: PropTypes.func.isRequired
}

export default Undo;
