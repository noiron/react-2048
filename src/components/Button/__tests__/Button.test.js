import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Button />', () => {
    const handleClick = jest.fn();
    const imageUrl = 'imageUrl';
    const imageDesc = 'desc';

    it('component render', () => {
        const button = renderer.create(
            <Button
                handleClick={handleClick}
                img={imageUrl}
                desc={imageDesc}
            />
        );
        expect(button.toJSON()).toMatchSnapshot();
    })

    it('button should has a image child', () => {
        const button = mount(
            <Button
                handleClick={handleClick}
                img={imageUrl}
                desc={imageDesc}
            />
        );
        // 按钮内部应该有一个用于展示的图片
        expect(button.find('img')).toHaveLength(1);
        expect(button.simulate('click'));
    })
})

