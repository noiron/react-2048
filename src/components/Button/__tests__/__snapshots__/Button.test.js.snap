// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component render 1`] = `
<div
  className={undefined}
  onClick={[Function]}
>
  <img
    alt="desc"
    src="imageUrl"
  />
</div>
`;
